src/axom/CMakeLists.txt

```diff
@@ -46,7 +46,7 @@ endif()
 # Combine all component object libraries into a unified library
 blt_add_library(NAME       axom
                 SOURCES    Axom.cpp
-                DEPENDS_ON ${AXOM_COMPONENTS_ENABLED})
+                DEPENDS_ON core ${AXOM_COMPONENTS_ENABLED})
```

Comment:
Apparently if you turned off all components (leaving just `core`), `core` was never added to the `axom` combined library.